meson.build

```diff
@@ -526,15 +533,37 @@ crypto = declare_dependency(
     include_directories: ssl_includes,
 )
 
-if ssl_link_args != [] and ssl_includes != []
+if wolfssl.found() or ssl_link_args != []
     have_ssl = true
     cdata.set('HAVE_LIBCRYPTO', 1)
-    cdata.set('OPENSSL_DHX', 1)
     cdata.set('UAM_DHX', 1)
+endif
+
+ssl_deps = []
+ssl_type = ''
+
+#if have_embedded_ssl and wolfssl.found()
+#    error('embedded SSL cannot be compiled when WolfSSL is installed')
+if have_embedded_ssl
+    ssl_type += 'built-in'
+    cdata.set('WOLFSSL_DHX', 1)
+    cdata.set('EMBEDDED_SSL', 1)
+elif wolfssl.found() and run_command(
+    '/bin/cat',
+    wolfssl.get_variable(pkgconfig: 'includedir') / 'wolfssl/options.h',
+    check: true,
```

Comment:
This seems like a somewhat crude way to detect the wolfssl library. I wonder if there is a more portable method?

Reply:
This is not detecting the WolfSSL library, it is checking if the WolfSSL package installed on the host OS has been compiled with the OpenSSL compatibility layer.